langchain/tools/base.py

```diff
@@ -123,3 +130,22 @@ async def arun(
                 observation, verbose=verbose_, color=color, name=self.name, **kwargs
             )
         return observation
+
+    def __call__(self, tool_input: str) -> str:
+        """Make tools callable with str input."""
+        return self.run(tool_input)
+
+    def run(self, tool_input: str, **kwargs: Any) -> str:
+        """Run the tool."""
+        if len(self.args) > 0:
```

Comment:
Maybe I'm missing something but isn't `self.args` of len 1 by default?

Reply:
subclasses can override it